### PR TITLE
Stop Signing over the `PartSetHeader` for Votes and remove it from the `Header`

### PR DESCRIPTION
**Pull request by evan-forbes**
## Description

This PR builds off of #441 and continues the work of #418 by not including the `PartSetHeader` in the [`CanonicalVote`](https://github.com/lazyledger/lazyledger-core/blob/2cea55b53e1f473b73530c761374d01c195f07e2/proto/tendermint/types/canonical.proto#L31-L38). It also removes the `PartSetHeader` from the `Header`'s hash by removing it entirely from both [`types.Header`](https://github.com/lazyledger/lazyledger-core/blob/2cea55b53e1f473b73530c761374d01c195f07e2/types/block.go#L508-L537) and [`tmproto.Header`](https://github.com/lazyledger/lazyledger-core/blob/2cea55b53e1f473b73530c761374d01c195f07e2/proto/tendermint/types/types.proto#L57-L82)

This PR can be merged into #441 or master after #441 is merged, whichever is most convenient for the reviewers

Closes: #456

### CODE REVIEW

File: proto/tendermint/types/types.proto [hunk 1]
```diff
@@ -63,23 +63,22 @@ message Header {
 
   // prev block info
   BlockID last_block_id = 5 [(gogoproto.nullable) = false];
-  PartSetHeader last_part_set_header = 6;
```
Comment (evan-forbes) [inline proto/tendermint/types/types.proto:66]: removal of the `LastPartSetHeader` field in the proto header

File: proto/tendermint/types/canonical.proto [hunk 2]
```diff
@@ -35,5 +35,4 @@ message CanonicalVote {
   CanonicalBlockID          block_id  = 4 [(gogoproto.customname) = "BlockID"];
   google.protobuf.Timestamp timestamp = 5 [(gogoproto.nullable) = false, (gogoproto.stdtime) = true];
   string                    chain_id  = 6 [(gogoproto.customname) = "ChainID"];
-  CanonicalPartSetHeader part_set_header = 7;
```
Comment (evan-forbes) [inline proto/tendermint/types/canonical.proto:38]: removal of the `PartSetHeader` from the `CanonicalVote`

File: types/canonical.go [hunk 3]
```diff
@@ -56,15 +56,13 @@ func CanonicalizeProposal(chainID string, proposal *tmproto.Proposal) tmproto.Ca
...
-		BlockID:       CanonicalizeBlockID(vote.BlockID),
-		Timestamp:     vote.Timestamp,
-		ChainID:       chainID,
-		PartSetHeader: &cppsh,
+		Type:      vote.Type,
+		Height:    vote.Height,       // encoded as sfixed64
+		Round:     int64(vote.Round), // encoded as sfixed64
+		BlockID:   CanonicalizeBlockID(vote.BlockID),
+		Timestamp: vote.Timestamp,
+		ChainID:   chainID,
```
Comment (evan-forbes) [inline types/canonical.go:65]: the removal of the `PartSetHeader` in the `CanonicalVote` allows for the preservation of the `PartSetHeader` in `tmproto.Vote`

**Review by evan-forbes — COMMENTED** [review]
There are not that many meaningful changes, so I pointed them out to hopefully save reviewers a bit of time.

File: types/vote_test.go [hunk 4]
```diff
@@ -294,3 +294,23 @@ func TestVoteProtobuf(t *testing.T) {
...
+	voteA := makeMockVote(1, 0, 0, pubKey.Address(), blockID, psh, currentTime).ToProto()
+	voteB := makeMockVote(1, 0, 0, pubKey.Address(), blockID, PartSetHeader{}, currentTime).ToProto()
+
+	err = pv.SignVote("test", voteA)
+	require.NoError(t, err)
+	err = pv.SignVote("test", voteB)
+	require.NoError(t, err)
+
+	require.Equal(t, voteA.Signature, voteB.Signature)
+
```
Comment (liamsi) [inline types/vote_test.go:315]: 👍🏼

**Review by liamsi — COMMENTED** [review]
Just FYI. This looks really good. I'd still like to go through each line which might take a while.

File: proto/tendermint/types/canonical.proto (hunk 2 above)
Comment (liamsi) [reply proto/tendermint/types/canonical.proto:38]: So the `CanonicalPartSetHeader` is still needed bc/ we have to sign over it in the proposal at least, right?

**Review by liamsi — APPROVED** [review]

File: proto/tendermint/types/canonical.proto (hunk 2 above)
Comment (Wondertan) [reply proto/tendermint/types/canonical.proto:38]: @liamsi, yes. I double-checked and `CanonicalPartSetHeader` is only used in the canonical proposal.

**Review by Wondertan — APPROVED** [review]
Was trying to find something, but seems totally good!

**Pull request merged**